xbmc/video/VideoEmbeddedImageFileLoader.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoEmbeddedImageFileLoader.h"
+
+#include "FileItem.h"
+#include "guilib/Texture.h"
+#include "utils/EmbeddedArt.h"
+#include "utils/StringUtils.h"
+#include "video/VideoInfoTag.h"
+#include "video/tags/IVideoInfoTagLoader.h"
+#include "video/tags/VideoInfoTagLoaderFactory.h"
+
+using namespace VIDEO;
+
+bool CVideoEmbeddedImageFileLoader::CanLoad(std::string specialType) const
+{
+  return StringUtils::StartsWith(specialType, "video_");
+}
+
+namespace
+{
+bool GetEmbeddedThumb(const std::string& path, const std::string& type, EmbeddedArt& art)
+{
+  CFileItem item(path, false);
+  std::unique_ptr<IVideoInfoTagLoader> pLoader;
+  pLoader.reset(CVideoInfoTagLoaderFactory::CreateLoader(item, ADDON::ScraperPtr(), false));
+  CVideoInfoTag tag;
+  std::vector<EmbeddedArt> artv;
+  if (pLoader)
+    pLoader->Load(tag, false, &artv);
+
+  for (const EmbeddedArt& it : artv)
+  {
+    if (it.m_type == type)
+    {
+      art = it;
+      break;
+    }
+  }
+  return !art.Empty();
+}
+} // namespace
+
+std::unique_ptr<CTexture> CVideoEmbeddedImageFileLoader::Load(
+    std::string specialType,
+    std::string filePath,
+    unsigned int preferredWidth,
+    unsigned int preferredHeight) const
+{
+  EmbeddedArt art;
+  if (GetEmbeddedThumb(filePath, specialType.substr(6), art))
```

Comment:
What's with the substring here?

Reply:
"Special types" for this loader are artwork types prefixed with `video_` like `video_poster` - this pulls off the prefix to pass along only the artwork type.

Reply:
Right, see my other comment about a possible future PR to use enums instead.